src/runtime_src/core/common/info_aie.cpp

```diff
@@ -63,6 +63,223 @@ addnodelist(const std::string& search_str, const std::string& node_str,
+  bd_info.push_back("double_buffer_enable");
+  bd_info.push_back("interleave_enable");
+  bd_info.push_back("interleave_count");
+  bd_info.push_back("fifo_mode");
+  bd_info.push_back("x_increment");
+  bd_info.push_back("x_wrap");
+  bd_info.push_back("x_offset");
+  bd_info.push_back("y_increment");
+  bd_info.push_back("y_wrap");
+  bd_info.push_back("y_offset");
+
+  return populate_bd_info(input_pt, pt_bd, bd_info);
+}
+
+// This function extract BD information for shim tiles of 1st generation aie architecture
+void 
+populate_shim_bd_info_aie(const boost::property_tree::ptree& input_pt, boost::property_tree::ptree& pt_bd){
+  std::vector<std::string> bd_info;
+
+  bd_info.push_back("base_address");
+  bd_info.push_back("length");
+  bd_info.push_back("lock_id");
+  bd_info.push_back("lock_acq_val");
+  bd_info.push_back("lock_acq_enable");
+  bd_info.push_back("lock_acq_val_enable");
+  bd_info.push_back("lock_rel_val");
+  bd_info.push_back("lock_rel_enable");
+  bd_info.push_back("lock_rel_val_enable");
+  bd_info.push_back("pkt_enable");
+  bd_info.push_back("pkt_id");
+  bd_info.push_back("pkt_type");
+  bd_info.push_back("valid_bd");
+  bd_info.push_back("use_next_bd");
+  bd_info.push_back("next_bd_id");
+  bd_info.push_back("smid");
+  bd_info.push_back("cache");
+  bd_info.push_back("qos");
+  bd_info.push_back("secure_access");
+  bd_info.push_back("burst_length");
+
+  return populate_bd_info(input_pt, pt_bd, bd_info);
+}
+
+// This function extract BD information for core tiles of 2nd generation aie architecture
+void 
+populate_core_bd_info_aieml(const boost::property_tree::ptree& input_pt, boost::property_tree::ptree& pt_bd){
+  std::vector<std::string> bd_info;
+
+  bd_info.push_back("base_address");
+  bd_info.push_back("length");
+  bd_info.push_back("lock_acq_id");
+  bd_info.push_back("lock_acq_val");
+  bd_info.push_back("lock_acq_enable");
+  bd_info.push_back("lock_rel_id");
+  bd_info.push_back("lock_rel_val");
+  bd_info.push_back("pkt_enable");
+  bd_info.push_back("pkt_id");
+  bd_info.push_back("pkt_type");
+  bd_info.push_back("valid_bd");
+  bd_info.push_back("use_next_bd");
+  bd_info.push_back("next_bd_id");
+  bd_info.push_back("tlast_suppress");
+  bd_info.push_back("out_of_order_bd_id");
+  bd_info.push_back("compression_enable");
+  bd_info.push_back("iteration_current");
+  bd_info.push_back("iteration_step_size");
+  bd_info.push_back("iteration_wrap");
+  bd_info.push_back("d0_step_size");
+  bd_info.push_back("d0_wrap");
+  bd_info.push_back("d1_step_size");
+  bd_info.push_back("d1_wrap");
+  bd_info.push_back("d2_step_size");
+
+  return populate_bd_info(input_pt, pt_bd, bd_info);
+}
+
+// This function extract BD information for shim tiles of 2nd generation aie architecture
+void 
+populate_shim_bd_info_aieml(const boost::property_tree::ptree& input_pt, boost::property_tree::ptree& pt_bd){
+  std::vector<std::string> bd_info;
+
+  bd_info.push_back("base_address");
```

Comment:
As you know the vector size already, dont use multiple push_back calls as it is going to reallocate your vector again and again.
Either allocate vector with predefined size and push_back, or use initlizer list.

Use in other places too.